extension.js

```diff
@@ -363,59 +363,66 @@ export default class MyExtension extends Extension {
     }
 
     enable() {
-        this._setting = this.getSettings()
-        this._checkRequirements();
-        this._loginManager = LoginManager.getLoginManager();
-        this.systemMenu = Main.panel.statusArea.quickSettings._system;
-
-        this._hibernateMenuItem = new PopupMenu.PopupMenuItem(__('Hibernate'));
-        this._hibernateMenuItemId = this._hibernateMenuItem.connect(
-            'activate',
-            () => this._onHibernateClicked()
-        );
-
-        this._hybridSleepMenuItem = new PopupMenu.PopupMenuItem(
-            __('Hybrid Sleep')
-        );
-        this._hybridSleepMenuItemId = this._hybridSleepMenuItem.connect(
-            'activate',
-            () => this._onHybridSleepClicked()
-        );
-
-        this._suspendThenHibernateMenuItem = new PopupMenu.PopupMenuItem(
-            __('Suspend then Hibernate')
-        );
-        this._suspendThenHibernateMenuItemId = this._suspendThenHibernateMenuItem.connect(
-            'activate',
-            () => this._onSuspendThenHibernateClicked()
-        );
-
-        let afterSuspendPosition =
-            this.systemMenu._systemItem.menu.numMenuItems - 5;
-
-        this.systemMenu._systemItem.menu.addMenuItem(
-            this._hybridSleepMenuItem,
-            afterSuspendPosition
-        );
-        this.systemMenu._systemItem.menu.addMenuItem(
-            this._hibernateMenuItem,
-            afterSuspendPosition
-        );
-        this.systemMenu._systemItem.menu.addMenuItem(
-            this._suspendThenHibernateMenuItem,
-            afterSuspendPosition
-        );
-
-        this._menuOpenStateChangedId = this.systemMenu._systemItem.menu.connect(
-            'open-state-changed',
-            (menu, open) => {
-                if (!open) return;
-                this._updateDefaults();
-                this._updateHaveHibernate();
-                this._updateHaveHybridSleep();
-                this._updateHaveSuspendThenHibernate();
+        var self = this;
```

Comment:
the => javascript operator AFAIK captures this, making the usual var self = this unnecessary. Also, people nowadays discourage using var in favour of let or const.

Reply:
Thanks for the feedback! I have updated the polling mechanism as per https://github.com/UbayGD/reboottouefi/blob/main/src/extension.js and as recommended in https://discourse.gnome.org/t/main-panel-statusarea-quicksettings-system-is-undefined/16827. Working on my machine.